### PR TITLE
Adding ability to json load array from CLI

### PR DESCRIPTION
**Pull request by bigmstone**
This answers #2903.

This PR allows a user to input an array of dicts or other complicated types of arrays from the CLI. It will first attempt to json load the parameter and if that fails it reverts back to separating the values by commas.

### CODE REVIEW

**Comment by emedvedev** [comment]
LGTM. A test would be nice to have for this feature :)

**Comment by Kami** [comment]
LGTM :+1: 

Let's please also update corresponding documentation - https://docs.stackstorm.com/reference/cli.html#specifying-parameters-with-type-array

**Comment by Kami** [comment]
And please also add changelog entry :)

**Comment by bigmstone** [comment]
@Kami added changelog entry. Will make PR for docs changes tomorrow.
@emedvedev added unit test to cover both branches. Oddly arrays weren't covered at all. That's resolved now.

Will take feedback on tests before merging. Thanks guys.

**Comment by emedvedev** [comment]
The tests look good, but I'd still wait for @Kami's feedback: he's our main stickler for testing. Pretty sure the only reason it why was me and not him who asked for tests is that he was just sleeping at the time. 😛

File: st2client/tests/unit/test_action.py [hunk 1]
```diff
@@ -223,6 +284,65 @@ def test_param_json_conversion(self):
     @mock.patch.object(
         httpclient.HTTPClient, 'post',
         mock.MagicMock(return_value=base.FakeResponse(json.dumps(LIVE_ACTION), 200, 'OK')))
+    def test_param_array_conversion(self):
```
Comment (Kami) [inline st2client/tests/unit/test_action.py]: Nice, LGTM :+1:

**Pull request merged**